### ISSUE TITLE
How to debug slow Drush?

### ISSUE THREAD

**Issue by gitressa** [drush-ops/drush#6181]
**Describe the bug**

For some reason, all commands take +2 minute ..., what is the best method of debugging this?

```
$ time vendor/bin/drush st -vvv
 [preflight] Config paths: /var/www/html/example.com/public_html/vendor/drush/drush/drush.yml
 [preflight] Alias paths: /var/www/html/example.com/public_html/web/drush/sites,/var/www/html/example.com/public_html/drush/sites
 [preflight] Commandfile search paths: /var/www/html/example.com/public_html/vendor/drush/drush/src
 [info] Starting bootstrap to max [0.68 sec, 9.67 MB]
 [debug] Trying to bootstrap as far as we can [0.68 sec, 9.67 MB]
 [info] Drush bootstrap phase: bootstrapDrupalRoot() [0.68 sec, 9.67 MB]
 [info] Change working directory to /var/www/html/example.com/public_html/web [0.68 sec, 9.67 MB]
 [info] Initialized Drupal 10.3.10 root directory at /var/www/html/example.com/public_html/web [0.68 sec, 9.72 MB]
 [info] Drush bootstrap phase: bootstrapDrupalSite() [0.68 sec, 10.17 MB]
 [debug] Could not find a Drush config file at sites/default/drush.yml. [0.69 sec, 10.41 MB]
 [info] Initialized Drupal site default at sites/default [0.69 sec, 10.41 MB]
 [info] Drush bootstrap phase: bootstrapDrupalConfiguration() [0.69 sec, 10.41 MB]
 [info] Drush bootstrap phase: bootstrapDrupalDatabase() [0.7 sec, 11.17 MB]
 [info] Successfully connected to the Drupal database. [0.7 sec, 11.17 MB]
 [info] Drush bootstrap phase: bootstrapDrupalFull() [0.7 sec, 11.17 MB]
 [debug] Start bootstrap of the Drupal Kernel. [0.7 sec, 11.17 MB]
 [debug] Finished bootstrap of the Drupal Kernel. [0.77 sec, 18.48 MB]
 [debug] Loading drupal module drush commands & etc. [0.77 sec, 18.51 MB]
 [debug] Found drush.services.yml for search_api_solr Drush commands [0.78 sec, 18.68 MB]
 [debug] Found drush.services.yml for search_api_solr_admin Drush commands [0.78 sec, 18.68 MB]
 [debug] Found drush.services.yml for token Drush commands [0.78 sec, 18.68 MB]
 [debug] Add a commandfile class: Drupal\search_api_solr\Commands\SearchApiSolrCommands [0.87 sec, 22.61 MB]
 [debug] Add a commandfile class: Drupal\search_api_solr_admin\Commands\SearchApiSolrAdminCommands [0.87 sec, 22.64 MB]
 [debug] Add a commandfile class: Drupal\token\Drush\Commands\TokenCommands [0.87 sec, 22.66 MB]
 [debug] Add a commandfile class: Drupal\pathauto\Commands\PathautoCommands [0.87 sec, 22.66 MB]
 [debug] Add a commandfile class: Drupal\xmlsitemap\Commands\XmlSitemapCommands [0.87 sec, 22.68 MB]
 [info] sql:query: SELECT VERSION();" [1.05 sec, 32.78 MB]
 [info] Executing: mysql --defaults-file=/tmp/drush_USjvmN --database=example_com --host=localhost --port=3306 --silent -A < /tmp/drush_YEVBJj [1.05 sec, 33.14 MB]

```

Here it hangs for ~2 minutes ... and then:

```
Drupal version   : 10.3.10                                                               
Site URI         : http://default                                                        
DB driver        : mysql                                                                 
DB hostname      : localhost                                                             
DB port          : 3306                                                                  
Database         : Connected                                                             
Drupal bootstrap : Successful                                                            
PHP binary       : /usr/bin/php8.3                                                       
PHP config       : /etc/php/8.3/cli/php.ini                                              
PHP OS           : Linux                                                                 
PHP version      : 8.3.14                                                                
Drush script     : /var/www/html/example.com/public_html/vendor/bin/drush.php         
Drush version    : 13.3.3.0                                                              
Drush temp       : /tmp                                                                  
Drush configs    : /var/www/html/example.com/public_html/vendor/drush/drush/drush.yml 
Install profile  : minimal                                                               
Drupal root      : /var/www/html/example.com/public_html/web                          
Site path        : sites/default                                                         
Files, Public    : sites/default/files                                                   
Files, Temp      : /tmp                                                                  

real  2m13,826s
user  0m0,257s
sys 0m0,166s
```

**To Reproduce**
Ran a Drush command, and had to wait more than 2 minutes for a result.

**Expected behavior**

**Actual behavior**

**Workaround**

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 13.3.3.0
| Drupal version? | 10.3.10
| PHP version     | 8.3.14
| OS?             | Debian

**Additional information**

**Comment by weitzman** [comment]
Try to run in the CLI what drush is running in that slow moment : `mysql --database=example_com --host=localhost --port=3306`. You will have to add your Db creds. See if it also waits 2 min. 

Any chance you are using mariadb? There was a recent release that started SSL checking - https://drupal.slack.com/archives/C62H9CWQM/p1734092627475469

**Comment by gitressa** [comment]
Thanks for a fast reply and suggestions. I tried your method of running the mysql command directly from the CLI, and I noticed that 3306 was not specified in the similar Drush call for some D7 sites ... So I tried leaving the port line empty in _settings.php_, and it started working.

Maybe port 3306 was never opened, so it tries to connect to port 3306 for two minutes, and then goes on to try the socket location? Anyway, I get instant feedback now, and everything works, so I'll close the issue. Thanks!
